### PR TITLE
Update shared components in three smaller Annotation components; convert to TS

### PR DESCRIPTION
**Pull request by lyzadanger**
More conversions! No user-visible changes here.

Part of #4860

### CODE REVIEW

File: src/sidebar/components/Annotation/AnnotationTimestamps.tsx [hunk 1]
```diff
@@ -7,14 +7,13 @@ import {
...
- * @prop {string} annotationCreated
- * @prop {string} annotationUpdated
- * @prop {string} [annotationUrl]
- * @prop {boolean} [withEditedTimestamp] - Should a timestamp for when this
- *   annotation was last edited be rendered?
- */
+export type AnnotationTimestampProps = {
+  annotationCreated: string;
+  annotationUpdated: string;
+  annotationUrl?: string;
```
Comment (robertknight) [inline src/sidebar/components/Annotation/AnnotationTimestamps.tsx:13]:
```suggestion
  annotationURL?: string;
```

File: src/sidebar/components/Annotation/AnnotationTimestamps.tsx [hunk 2]
```diff
@@ -7,14 +7,13 @@ import {
...
+  /** Should a timestamp for when this annotation was last updated be rendered? */
```
Comment (robertknight) [inline src/sidebar/components/Annotation/AnnotationTimestamps.tsx:14]: The phrasing here is a bit awkward. Something like "Display a relative last-updated timestamp" might be better.

File: src/sidebar/components/Annotation/AnnotationLicense.tsx [hunk 3]
```diff
@@ -7,13 +11,14 @@ export default function AnnotationLicense() {
...
         href="http://creativecommons.org/publicdomain/zero/1.0/"
         target="_blank"
         title="View more information about the Creative Commons Public Domain dedication"
       >
-        <Icon name="cc-std" classes="text-tiny" />
-        <Icon name="cc-zero" classes="ml-px text-tiny" />
+        <CcStdIcon className="w-em h-em text-tiny" />
+        <CcZeroIcon className="w-em h-em ml-px text-tiny" />
         <div className="ml-1">
           Annotations can be freely reused by anyone for any purpose.
```
Comment (robertknight) [inline src/sidebar/components/Annotation/AnnotationLicense.tsx:23]: This text currently has slightly weak link-ness cues as there is only a hover color and no underline. This was true before the conversion though.

**Review by robertknight — APPROVED** [review]

File: src/sidebar/components/Annotation/AnnotationLicense.tsx (hunk 3 above)
Comment (lyzadanger) [reply src/sidebar/components/Annotation/AnnotationLicense.tsx:23]: Yeah I know...

**Pull request merged**